src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1114,6 +1124,9 @@ public DocumentId AddFile(string fullPath, SourceCodeKind sourceCodeKind, Immuta
                         throw new ArgumentException($"'{fullPath}' has already been added to this project.", nameof(fullPath));
                     }
 
+                    // Adding a file messes up the ordering and does not gaurantee any kind of order, so invalidate any ordered files in batch.
```

Comment:
I guess I would have imagined this would just update _orderedFilesInBatch to stick it at the end. This otherwise seems...surprising.

Reply:
I guess we could do that, but I would argue that calling AddSourceFile/RemoveSourceFile does not guarantee order therefore we should try not to force any kind of order when we do those actions. But, if you think it's ok, then I guess it's alright to add/remove a file from the batch, but we would allocate again.